braindecode/datasets/base.py

```diff
@@ -60,13 +60,13 @@ class WindowsDataset(BaseDataset):
     def __init__(self, windows, description):
         self.windows = windows
         self.description = description
+        self.index_info_cols = [
```

Comment:
Please move the variable back into the function. Reason is that variables should in general be in smallest scope, see  for example https://softwareengineering.stackexchange.com/a/388055 for soem reasoning on this. I expect recreating list in method to not have any noticeable impact on performance. If that is wrong, e.g. performance tests show otherwise, we can move it out then. If we need the variable outside the function for other reasons, as you mentioned as a future possibility, we will move it outside at that point.